packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -225,6 +236,11 @@ class MediaQueryData {
       gestureSettings = DeviceGestureSettings.fromView(view),
       displayFeatures = view.displayFeatures;
 
+  static TextScaler _textScalerFromView(ui.FlutterView view, MediaQueryData? platformData) {
+    final double scaleFactor = platformData?.textScaleFactor ?? view.platformDispatcher.textScaleFactor;
```

Comment:
Yes, if the decision is to hard-code the scaling curve. Otherwise the `TextScaler` class will have to be moved to `dart:ui`, in order to access the platform API or AndroidX API.